VideoApi/VideoApi.Events/Handlers/Core/EventHandlerBase.cs

```diff
@@ -46,7 +46,7 @@ public virtual async Task HandleAsync(CallbackEvent callbackEvent)
                 .SingleOrDefault(x => x.Id == callbackEvent.ParticipantId);
 
             SourceEndpoint = SourceConference.GetEndpoints().SingleOrDefault(x => x.Id == callbackEvent.ParticipantId);
-            SourceRoom = SourceConference.Rooms.SingleOrDefault(x => x.Id == callbackEvent.RoomId);
+            SourceInterpreterRoom = SourceConference.Rooms.SingleOrDefault(x => x is InterpreterRoom && x.Id == callbackEvent.ParticipantRoomId) as InterpreterRoom;
```

Comment:
OfType would remove the need for this and the casting at the end

Reply:
Thanks